fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergMetaCache.java

```diff
@@ -0,0 +1,118 @@
+
+    public IcebergMetaCache(Executor executor) {
+        CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+        cacheBuilder.expireAfterWrite(Config.iceberg_meta_cache_ttl_s, SECONDS);
+        cacheBuilder.maximumSize(MAX_TABLE_CACHE_SIZE);
+        tableStatsCache = cacheBuilder.build(asyncReloading(
+                new CacheLoader<IcebergTableKey, IcebergFileStats>() {
+                    @Override
+                    public IcebergFileStats load(IcebergTableKey key) throws Exception {
+                        Table table = getTable(key);
+                        return getIcebergFileStats(table, getCalculator(table));
+                    }
+                }, executor));
+    }
+
+    public Statistics getTableStatistics(IcebergTableKey key,
+                                         Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
+        IcebergFileStats icebergFileStats;
+        try {
+            icebergFileStats = tableStatsCache.get(key);
+        } catch (ExecutionException e) {
+            LOG.warn("Get iceberg file stats from cache with failure!", e);
+            Table table = getTable(key);
+            icebergFileStats = getIcebergFileStats(table, getCalculator(table));
+        }
+        Table table = getTable(key);
+        IcebergTableStatisticCalculator calculator = getCalculator(table);
+        List<Types.NestedField> columns = table.schema().columns();
+        return calculator.makeTableStatistics(icebergFileStats, columns, colRefToColumnMetaMap);
+    }
+
+    public List<ColumnStatistic> getColumnStatistics(IcebergTableKey key,
+                                                     Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
+        IcebergFileStats icebergFileStats;
+        try {
+            icebergFileStats = tableStatsCache.get(key);
+        } catch (ExecutionException e) {
+            LOG.warn("Get iceberg file stats from cache with failure!", e);
+            Table table = getTable(key);
+            icebergFileStats = getIcebergFileStats(table, getCalculator(table));
+        }
+        Table table = getTable(key);
+        IcebergTableStatisticCalculator calculator = getCalculator(table);
+        List<Types.NestedField> columns = table.schema().columns();
+        return calculator.makeColumnStatistics(icebergFileStats, columns, colRefToColumnMetaMap);
+    }
+
+    @VisibleForTesting
+    protected IcebergFileStats getIcebergFileStats(Table table,
+                                                IcebergTableStatisticCalculator calculator) {
+        IcebergFileStats icebergFileStats = calculator.generateIcebergFileStats(
+                new ArrayList<>(), table.schema().columns()
+        );
+        return icebergFileStats;
+    }
+
+    @VisibleForTesting
+    protected IcebergTableStatisticCalculator getCalculator(Table table) {
+        IcebergTableStatisticCalculator calculator = new IcebergTableStatisticCalculator(table);
```

Comment:
I think the definition and calling relationship of IcebergTableStatisticCalulator can be changed again, the redundant code is a bit.